### PR TITLE
feat(events): integrate generic item event sources

### PR DESCRIPTION
**Pull request by donni106**
Example settings for st-md participation project entries:

```
"eventCalendar": {
  "genericItemEventSources": [
    {
      "genericType": "ParticipationProject",
      "filterTypes": [
        "Veranstaltung"
      ],
      "filterStatuses": [
        "active",
        "announced"
      ]
    }
  ]
}
```

SVA-1751

### CODE REVIEW

File: src/components/widgets/EventWidget.tsx [hunk 1]
```diff
@@ -19,8 +19,11 @@ const today = moment().format('YYYY-MM-DD');
 export const EventWidget = ({ text, additionalProps, widgetStyle }: WidgetProps) => {
   const navigation = useNavigation();
   const { globalSettings } = useContext(SettingsContext);
-  const { hdvt = {} } = globalSettings;
+  const { hdvt = {}, settings = {} } = globalSettings;
   const { events: showVolunteerEvents = false } = hdvt as { events?: boolean };
+  const genericItemEventSources =
+    (settings as { eventCalendar?: { genericItemEventSources?: [] } }).eventCalendar
+      ?.genericItemEventSources || [];
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/widgets/EventWidget.tsx]: Good catch. I replaced the empty-tuple assertion with `GenericItemEventSource[]`, so tenant configuration retains its actual array shape and is type checked. Addressed in b934aed93.

File: src/hooks/genericItemEvents.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      },
+      enabled: queryEnabled
+    }))
+  );
+
+  const data = useMemo(() => {
+    if (!queryEnabled) return EMPTY_DATA;
+    const merged = validSources.flatMap((source) => {
+      const index = genericTypes.indexOf(`${source.genericType}`);
+      const result = queries[index]?.data as Record<string, unknown[]> | undefined;
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/hooks/genericItemEvents.ts:50]: Good catch. Sources are now normalized once by trimming `genericType`; the normalized value is used for deduplicated query keys, requests, source/result matching, and parsing. I also added a regression test for surrounding whitespace. Addressed in b934aed93.
